CorsixTH/Lua/app.lua

```diff
@@ -1517,6 +1517,26 @@ function App:quickLoad()
   end
 end
 
+--! Function to check the savegame version matches or is older than the installation of CorsixTH
+--!param save_version (num)
+--!param save_release (string) Game's human readable game version
+--!return true if compatible, otherwise false
+function App:checkCompatibility(save_version, save_release)
+  local app_version = self.savegame_version
+  local version_name = self:getVersion(app_version)
+  local save_release_ver = save_release or "Trunk" -- For compatibility
+  local app_version_humanize = tostring(version_name .. " [" .. app_version .. "]")
+  local save_version_humanize = tostring(save_release_ver .. " [" .. save_version .. "]")
+
+  if app_version >= save_version or self.config.debug then
+    return true
+  else
+    local err = _S.errors.compatibility_error:format(save_version_humanize, app_version_humanize)
```

Comment:
Why do you compute strings to report as error before you decided there is an error?

Reply:
The "error" happens here at 1531 if false.

Reply:
yes, so computing the text can be done after line 1533